### ISSUE TITLE
Incorrect calculation of months over a long interval

### ISSUE THREAD

**Issue by bostonaholic** [andrewmcveigh/cljs-time#34]
I have updated the `cljs-time.core-test/long-interval-test` to show it failing.

```
(deftest long-interval-test
  (let [start (date-time 2015 5 6)
        end (date-time 2024 5 5)
        i (interval start end)]
    (is (= 108 (in-months i)))))
```

```
FAIL in (cljs-time.core-test/long-interval-test) (:)
expected: (= 108 (in-months i))
  actual: (not (= 108 102))
```

Also, unrelated (but maybe not), the test suite does not pass for me aside from the one above. I get 3 failures, all the same from each of the different compilations.

```
FAIL in (cljs-time.core-test/test-today-default) (:)
expected: (= (local-date 2013 4 20) (do-at (to-default-time-zone (date-time 2013 4 20)) (today)))
  actual: (not (= #<20130420> #<20130419>))
```

**Comment by bostonaholic** [comment]
This could be related to #32

**Comment by andrewmcveigh** [comment]
That was pretty slack of me to not actually test in-months. Fixed in `0.3.7`.

**Comment by bostonaholic** [comment]
:+1:
